### ISSUE TITLE
CORE/PERSIST interoperability profile for provenance-bearing persona-state revision

### ISSUE THREAD

**Issue by crates** [entif-ai/rosetta#1592]
## Summary

Create a public **CORE/PERSIST interoperability profile** that maps the paper's controlled persona-state revision and stress-test records into existing Rosetta artifacts without changing Rosetta Core or standardizing one private persona-learning algorithm.

The smallest honest artifact is a SchemaPack plus a thin JSON/JSONL adapter:

```text
dialogue turn + prior state
  -> turn-local evidence observation
  -> Commit / Defer / Ignore routing action
  -> optional gated belief revision
  -> successor learned-persona-state snapshot
  -> PERSIST response/state/action evaluations
  -> receipt-bundle tapestry
```

This issue is a method-specific consumer of #1476 (learned persona state), #1510 (persona drift evaluation record), #158 (receipt family), #994 (write admission), and #1296 (identity-sensitive operation gate). It must not duplicate or broaden those contracts.

## Research basis

Paper:

- Youyuan Zhang, Siyuan Li, Fangming Liu, and Jing Li, **"Toward Robust Personalized Alignment for LLMs: Mitigating Persona Drift in Multi-Turn Dialogue"**, arXiv:2609.12373, submitted 11 September 2026 and accepted to Findings of EMNLP 2026: https://arxiv.org/abs/2609.12373
- Full text: https://arxiv.org/pdf/2609.12373
- Canonical Entif comparison sources:
  - https://entif.ai/tags/research/2026/09/10/babel-fish-alien-mind/
  - https://entif.ai/tags/research/2026/09/09/rosetta-pasigraphy-protocol/

No attributable public code repository or dataset release URL is linked from the arXiv record or paper as of this issue's creation. The first phase therefore uses specification-derived schemas and synthetic fixtures only. No paper implementation should be copied or represented as licensed for reuse without a later explicit release and license review.

The paper's relevant constructs are unusually concrete:

- slot-factorized persistent belief (b_t^k)
- distinct persistent `UNK` and turn-local null `⊥`
- turn-local evidence distribution `p_tilde`
- relevance, ambiguity, conflict, and prior-confidence descriptors
- explicit `COMMIT | DEFER | IGNORE` update action
- a revision gate that is zero outside `COMMIT`
- KL-regularized successor belief
- separate `RESPOND | CLARIFY` discourse action
- PERSIST phases: Anchoring, Fluctuation, Confrontation
- six social-influence strategies
- response-, belief-, action-, and clarification-level metrics from different evidence sources

Mechanistic results make the integration falsifiable: removing executable update actions raises unsupported `COMMIT` from 11.3% to 22.9%; removing clarification raises under-clarification from 18.4% to 42.7%; and the matched structure-only control underperforms executable revision on both alignment and conflict robustness.

## Why this belongs in Rosetta

Rosetta Core already requires immutable Observations, explicit Actions, Evaluations, lineage, content addressing, and Receipts. Issues #1476 and #1510 define public boundaries for learned persona-state snapshots and persona-drift evaluation records. CORE supplies a concrete external method with which to test whether those boundaries can preserve:

1. what a turn suggested;
2. whether that suggestion was relevant, ambiguous, or in conflict;
3. what update action was selected;
4. whether durable revision was authorized;
5. what state changed, if anything;
6. whether the system clarified or responded;
7. how a later evaluator scored each object; and
8. what a downstream consumer may rely on.

The integration is useful precisely because these claims must remain non-collapsed. A relevant observation is not permission to persist it. A `DEFER` result is not a future `COMMIT`. A drift score is not mutation authority. A receipt proves that a declared protocol ran over named artifacts; it does not prove that a user's true preference was discovered.

## Concept and primitive crosswalk

| CORE / PERSIST construct | Rosetta representation | Mapping status and guardrail |
| --- | --- | --- |
| User turn and dialogue span | `rosetta.observation` | Exact reuse; preserve source text or a privacy-safe reference plus selector |
| Prior persona belief | #1476 learned persona-state snapshot reference | Partial; CORE's slot distribution is a method-specific payload/profile, not a new Core kind |
| `UNK` | Profile vocabulary value `unknown_persistent_value` | Exact method mapping; must not collapse into missing data |
| Turn-local `⊥` | Evidence observation status `no_usable_slot_evidence` | Exact method mapping; distinct from persistent `UNK` |
| Evidence logits/distribution | Derived `rosetta.observation` or evaluation input artifact | Partial; retain model/checkpoint and normalization provenance |
| relevance / ambiguity / conflict / prior confidence | `rosetta.evaluation` metrics | Exact reuse with metric-profile/version refs |
| `COMMIT | DEFER | IGNORE` | `rosetta.action` with profile vocabulary | Exact method mapping; no new Rosetta Core action taxonomy |
| Revision gate and method version | Action parameters plus policy/method refs | Partial; publishing values is optional and privacy/policy constrained |
| Successor belief | New #1476 snapshot with `prev` / supersession linkage | Exact public boundary; never mutate the prior snapshot in place |
| `RESPOND | CLARIFY` | Separate `rosetta.action` | Exact reuse; must not be inferred only from generated prose |
| PERSIST phase / strategy / gold behavior | Versioned benchmark observation/fixture metadata | Method-specific profile |
| ARUS, BCD, drift, FCR, APR, action accuracy | #1510-compatible `rosetta.evaluation` records | Partial; keep metric source and measured subject explicit |
| Benchmark run | `rosetta.run` | Exact reuse |
| Attested execution and bounded claim | `rosetta.receipt` plus `rosetta.tapestry` closure | Exact reuse; receipt is not truth or user-consent authority |

## Recommended collaboration artifact

Use a **SchemaPack / interoperability profile**, not a new Core primitive and not a general-purpose PersonaPack.

Proposed identity: `core-persist-interoperability@0.1.0-candidate`.

The pack should contain:

- vocabulary for method-specific states, actions, phases, and strategies;
- JSON Schemas for an evidence record, routing record, revision record, and PERSIST evaluation record;
- a loss-aware mapping to #1476 and #1510;
- synthetic JSONL fixtures and expected receipt-bundle closures;
- a thin adapter interface for future paper outputs;
- a researcher-readable crosswalk and limitations note.

This is appropriate because the paper exposes stable record boundaries, but no public implementation or universal ontology. A Pack can preserve the mapping without claiming that CORE defines Rosetta semantics or that Rosetta implements CORE.

## Proposed architecture

1. **Input boundary**
   - ingest a privacy-minimized dialogue-turn reference, prior snapshot CID, slot ID, model/checkpoint identifier, and method-profile version;
   - reject raw sensitive persona content by default in public fixtures.

2. **Evidence record**
   - record candidate value vocabulary, explicit `UNK` and `⊥` treatment, turn-local scores/distribution, and descriptor metrics;
   - link every derived field to the turn observation and prior snapshot.

3. **Routing action**
   - record `COMMIT`, `DEFER`, or `IGNORE`, the method/profile version, inputs used, and whether the action is advisory or write-eligible;
   - require `COMMIT` before the adapter may propose a successor snapshot.

4. **Write-admission boundary**
   - route every durable successor snapshot through #994;
   - route identity/person-model-sensitive handling through #1296;
   - fail closed to no state mutation on absent authority, ambiguous scope, frozen/opt-out state, stale policy, or incomplete provenance.

5. **Revision record**
   - on admitted `COMMIT`, create a new snapshot linked to its predecessor;
   - preserve distributional state when disclosed, or a declared loss profile when projected into a simpler public snapshot;
   - on `DEFER` or `IGNORE`, retain the previous snapshot CID unchanged.

6. **Discourse action and response**
   - record `RESPOND` or `CLARIFY` independently of the update action;
   - permit a deferred observation to inform the current turn without persisting it.

7. **Evaluation and closure**
   - emit source-separated response, state, action, and clarification evaluations;
   - compile the exact subjects, policies, method profiles, outputs, and evaluations into a receipt-bundle tapestry.

## Translation and interoperability design

Define four candidate public records:

- `core_persist.evidence_record`
- `core_persist.routing_record`
- `core_persist.revision_record`
- `core_persist.evaluation_record`

These names are proposed Pack-local identifiers, not Rosetta Core kinds.

Required loss-awareness:

- `UNK` and `⊥` MUST remain distinguishable.
- Missing distribution values MUST be marked `not_disclosed`, `not_recorded`, or `not_applicable`; they MUST NOT become zero.
- A scalar MAP value MUST declare that distributional information was dropped.
- `DEFER` MUST NOT create an implied durable pending memory.
- `IGNORE` MUST remain distinguishable from irrelevant input filtered before routing.
- An evaluation verdict MUST name its measured subject and metric source.
- PERSIST response-only evaluation MUST NOT imply that internal routing/state conformed to CORE.
- Successor-state creation MUST reference both the prior snapshot and the admitted routing/write decision.

Adapter sketch:

```ts
type CorePersistAdapter = {
  mapEvidence(input: unknown, profile: MethodProfile): EvidenceRecord;
  mapRouting(input: unknown, evidenceRef: string): RoutingRecord;
  mapRevision(input: unknown, routingRef: string, priorStateRef: string): RevisionRecord;
  mapEvaluation(input: unknown, subjects: string[]): EvaluationRecord[];
  validateClosure(refs: string[]): VerificationReport;
};
```

## Repository and implementation impact

Proposed locations, subject to repository review:

- `packs/core-persist/manifest.json`
- `packs/core-persist/schema/*.schema.json`
- `packs/core-persist/vocab/*.json`
- `packs/core-persist/mappings/learned-persona-state.json`
- `packs/core-persist/mappings/persona-drift-evaluation.json`
- `packs/core-persist/fixtures/*.jsonl`
- `packs/core-persist/README.md`
- `packages/rosetta-schemas`: register only accepted Pack schemas
- `packages/rosetta-receipts`: reuse receipt and tapestry closure behavior; do not add a parallel receipt engine
- `packages/rosetta-guard`: add only parse-only fixture checks needed to prove write-admission handoff

Before accepting names, run semantic-governance review and record that all `core_persist.*` identifiers are Pack-local extensions.

### Phase 0: research and authority lock

- confirm the paper-version checksum and citation;
- request author review of the crosswalk;
- confirm whether code/data will be released and under what license;
- resolve Pack-local names against the Core Spine, #1476, #1510, and semantic audit.

### Phase 1: schemas and loss map

- define the four Pack-local schemas;
- define explicit missing/redacted/not-disclosed states;
- publish the primitive crosswalk and projection loss table;
- add schema-valid positive and negative fixtures.

### Phase 2: synthetic adapter

- implement a thin JSON/JSONL adapter against synthetic CORE-shaped traces;
- emit Rosetta Run, Observation, Action, Evaluation, Receipt, and Tapestry references;
- prove `DEFER` and `IGNORE` cannot create successor-state snapshots.

### Phase 3: PERSIST evaluation profile

- encode synthetic Anchoring -> Fluctuation -> Confrontation trajectories;
- preserve phase, strategy, stressed axiom/slot, conflict and ambiguity flags, target behavior, and metric source;
- produce #1510-compatible evaluation records without treating response scores as state evidence.

### Phase 4: author-reviewed real-data interoperability

- proceed only after an attributable public data/code release and license review;
- map one released trajectory end to end;
- document divergences and amend the Pack rather than silently coercing fields.

## Detailed task checklist

- [ ] Record paper/version/hash and source URLs.
- [ ] Confirm no existing issue already owns this exact CORE/PERSIST adapter.
- [ ] Review #1476, #1510, #158, #994, #1296, and #807.
- [ ] Define Pack manifest and namespace.
- [ ] Define evidence, routing, revision, and evaluation schemas.
- [ ] Define `UNK` versus `⊥` conformance tests.
- [ ] Define `COMMIT | DEFER | IGNORE` vocabulary.
- [ ] Define `RESPOND | CLARIFY` vocabulary independently.
- [ ] Define Anchoring / Fluctuation / Confrontation and six-strategy fixtures.
- [ ] Define method/checkpoint/prompt/metric-profile provenance.
- [ ] Define projection loss markers.
- [ ] Define write-admission and identity-sensitive gate handoffs.
- [ ] Add synthetic fixtures for admitted commit, deferred ambiguity, ignored local cue, explicit reversal, and pressure-first initialization.
- [ ] Add invalid fixtures for in-place mutation, implicit deferred persistence, response/state metric collapse, and receipt-as-truth claims.
- [ ] Emit and verify receipt-bundle tapestry closure.
- [ ] Run semantic-governance checks before schema registration.
- [ ] Request researcher review before mapping any released implementation artifact.

## Acceptance criteria and verification

- [ ] All Pack schemas validate deterministic positive fixtures and reject negative fixtures.
- [ ] `UNK` and `⊥` survive a round trip without collision.
- [ ] `DEFER` and `IGNORE` preserve the prior state CID and cannot emit a successor snapshot.
- [ ] An admitted `COMMIT` creates a new immutable snapshot linked to the prior snapshot and write-admission evidence.
- [ ] Frozen, opt-out, scope-mismatched, unauthorized, or identity-sensitive ambiguous cases fail closed.
- [ ] Response, belief, routing-action, clarification, and drift metrics identify distinct subjects and sources.
- [ ] A response-only PERSIST record cannot claim CORE-internal conformance.
- [ ] Receipt verification reconstructs the declared run closure without claiming preference truth, user consent, or downstream authorization.
- [ ] Synthetic fixtures reproduce the paper's boundary cases without containing real user data.
- [ ] Mapping documentation clearly labels paper facts, Rosetta reuse, Pack proposals, and unresolved questions.
- [ ] No new Rosetta Core kind is introduced.

## Risks, constraints, and licensing

- The paper uses structured benchmark slots; open-ended and cross-slot preferences remain outside demonstrated scope.
- Conservative routing may delay genuine but weakly expressed changes.
- PERSIST primarily measures post-anchor robustness and cannot determine whether a real user's long-term preference truly changed.
- Model-assisted data construction and judge-based response metrics require explicit source and reliability metadata.
- Public persona records can create privacy, profiling, manipulation, and identity risks; minimize content, require scope/consent controls, and consume #1296.
- The paper reports no public code/data URL in the inspected version. Do not vendor, reproduce, or imply licensing of an implementation until a release is attributable and its license is reviewed.
- A Rosetta receipt attests execution and artifact relationships, not psychological truth or author endorsement.

## Open questions for the researchers

1. Will CORE and PERSIST code/data be released, and under what license?
2. Which intermediate objects are actually logged in the reference implementation: full slot distributions, descriptors, routing probabilities, gate values, prior/successor snapshots, and discourse actions?
3. Is `DEFER` intentionally non-durable in every implementation path, or are there experimental pending-evidence variants?
4. What stable identifier should bind a slot across changing semi-open value vocabularies?
5. Which metrics are considered valid for cross-model comparison versus only within one backbone/configuration?
6. Can PERSIST expose a privacy-safe canonical trajectory for interoperability testing?
7. How should explicit user correction, deletion, and opt-out be represented relative to learned revision?
8. Which failure cases most often confuse `DEFER` and `IGNORE`, and should interoperability preserve verifier uncertainty?
9. Would the authors review a Rosetta mapping that keeps algorithm internals optional while standardizing artifacts and provenance?

## Non-goals

- Reimplementing or claiming endorsement of CORE.
- Defining a universal persona ontology or universal update threshold.
- Publishing private weights, logits, user profiles, or sensitive dialogue.
- Treating persona state as immutable truth about a person.
- Replacing #1476, #1510, #994, #1296, or Rosetta Core.
- Making drift evaluation authorize state mutation.
- Inferring hidden model state from responses.
- Claiming PERSIST coverage of cross-session retrieval corruption, third-party updates, or all real-world social influence.
- Adding a production personalization service in this issue.

## Collaboration path

A practical first collaboration is a 60-minute schema review around one synthetic PERSIST trajectory. Rosetta maintainers can supply the Pack skeleton, loss table, receipt closure, and conformance fixtures; the paper authors can validate terminology, field meanings, benchmark boundaries, and which intermediate records can be released safely. A later phase can map one attributable public trace after code/data licensing is clear.

Participation would be voluntary and non-commercial. The issue does not imply author endorsement or organizational partnership.

## Source references

- Paper abstract and metadata: https://arxiv.org/abs/2609.12373
- Paper full text: https://arxiv.org/pdf/2609.12373
- We Had the Seeds of a Babel Fish: https://entif.ai/tags/research/2026/09/10/babel-fish-alien-mind/
- Rosetta Pasigraphy Protocol: https://entif.ai/tags/research/2026/09/09/rosetta-pasigraphy-protocol/
- Rosetta repository: https://github.com/entif-ai/rosetta/
- Learned persona state contract: https://github.com/entif-ai/rosetta/issues/1476
- Persona drift evaluation record: https://github.com/entif-ai/rosetta/issues/1510
- Memory plane law: https://github.com/entif-ai/rosetta/issues/807
- Write-admission gate: https://github.com/entif-ai/rosetta/issues/994
- Identity-sensitive operation gate: https://github.com/entif-ai/rosetta/issues/1296
- Canonical receipt family: https://github.com/entif-ai/rosetta/issues/158
